server/server/src/upload_plugin.rs

```diff
@@ -0,0 +1,73 @@
+use std::fs;
+
+use actix_multipart::form::{tempfile::TempFile, MultipartForm};
+use actix_web::{
+    post,
+    web::{self, Data},
+    HttpResponse,
+};
+
+use base64::{prelude::BASE64_STANDARD, Engine};
+use serde::Deserialize;
+use serde_json::json;
+use service::{
+    backend_plugin::plugin_provider::{
+        BindPluginInput, PluginInstance, PluginType, PluginVariantType,
+    },
+    settings::Settings,
+    static_files::{StaticFileCategory, StaticFileService},
+};
+
+pub fn config_upload_plugin(cfg: &mut web::ServiceConfig) {
+    cfg.service(plugin);
+}
+
+// Should be in plugin_provider
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct UrlParams {
+    r#type: PluginType,
+    variant_type: PluginVariantType,
+    code: String,
+}
+
+// Stub
+#[derive(Deserialize)]
+pub struct PluginConfig {}
+
+#[derive(MultipartForm)]
+pub struct UploadPlugin {
+    #[multipart(rename = "files")]
+    pub file: TempFile,
+    pub config: actix_multipart::form::json::Json<PluginConfig>,
+}
+
+#[post("/plugin")]
+async fn plugin(
+    MultipartForm(UploadPlugin { file, .. }): MultipartForm<UploadPlugin>,
+    url_params: web::Query<UrlParams>,
+    settings: Data<Settings>,
+) -> HttpResponse {
+    let file_service = StaticFileService::new(&settings.server.base_dir).unwrap();
+
+    let static_file = file_service
+        .move_temp_file(file, &StaticFileCategory::Temporary, None)
+        .unwrap();
+
+    let bundle_base64 = BASE64_STANDARD.encode(fs::read(static_file.to_path_buf()).unwrap());
```

Comment:
Just a random thought, but do we actually need to BASE64 this? Can't it just be a string?
Not sure what the advantages/disadvantages would be though...

Reply:
Discussed in person, happy to keep base64